src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        FigureSupplier figureSupplier = new FigureSupplier(colorSupplier);
+        for (int i = 0; i < 6; i++) {
+            AbstractFigure figure = (AbstractFigure) figureSupplier.getRandomFigure();
```

Comment:
not implement the task correctly

Reply:
getRandomFigure should return AbstractFigure, no casting should be involved here